pkg/daemon/daemon.go

```diff
@@ -1406,7 +1406,20 @@ func checkV3Units(units []ign3types.Unit) error {
 	for _, u := range units {
 		for j := range u.Dropins {
 			path := filepath.Join(pathSystemd, u.Name+".d", u.Dropins[j].Name)
-			if err := checkFileContentsAndMode(path, []byte(*u.Dropins[j].Contents), defaultFilePermissions); err != nil {
+
+			var content string
+			if u.Dropins[j].Contents == nil {
+				content = ""
+			} else {
+				content = *u.Dropins[j].Contents
+			}
+
+			// Backwards compatibility check: the new behavior is to noop when a drop-in for zero length.
+			// However, to maintain backwards compatibility, we allow existing zero length files to exist.
+			if err := checkFileContentsAndMode(path, []byte(content), defaultFilePermissions); err != nil {
```

Comment:
This doesn't return a IsNotExist since it Wraps the error as follows:

```
	fi, err := os.Lstat(filePath)
	if err != nil {
		return errors.Wrapf(err, "could not stat file %q", filePath)
	}
```
A separate if above this to check for empty file is probably needed